### PR TITLE
assumeRole implementation

### PR DESCRIPTION
**Pull request by kr7ysztof**
https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRole.html

To obtain credential for a role user has to belong to the role in a keycloak (the information is in JWT returned by keycloak and provided to roku-sts)

### CODE REVIEW

File: src/main/scala/com/ing/wbaa/rokku/sts/api/STSApi.scala [hunk 1]
```diff
@@ -33,7 +33,15 @@ trait STSApi extends LazyLogging with TokenXML {
     }
   }
 
+  private val assumeRoleInputs = {
+    val inputList = ('RoleArn, 'RoleSessionName, "DurationSeconds".as[Int].?)
+    (parameters(inputList) | formFields(inputList)).tmap(t =>
+      t.copy(_1 = AwsRoleArn(t._1), _3 = parseDurationSeconds(t._3))
```
Comment (arempter) [inline src/main/scala/com/ing/wbaa/rokku/sts/api/STSApi.scala]: not very readable

File: README.md [hunk 2]
```diff
@@ -98,6 +100,30 @@ returns:
 </GetSessionTokenResponse>
 ```
 
+```http://localhost:12345?Action=AssumeRole&RoleArn=arn:aws:iam::account-id:role/admin&RoleSessionName=test```
```
[the remark this answers is not in the document]
Comment (kr7ysztof) [reply README.md]: added

File: src/main/scala/com/ing/wbaa/rokku/sts/api/xml/TokenXML.scala [hunk 3]
```diff
@@ -28,4 +55,23 @@ trait TokenXML extends ScalaXmlSupport {
...
+
+  /**
+   * aws doc:
+   * The Amazon Resource Name (ARN) and the assumed role ID,
+   * which are identifiers that you can use to refer to the resulting temporary security credentials.
+   * For example, you can reference these credentials as a principal in a resource-based policy
+   * by using the ARN or assumed role ID.
+   * The ARN and ID include the RoleSessionName that you specified when you called AssumeRole
+   *
+   * @param roleArn         - arn role e.g. arn:airlock:sts...
```
[the remark this answers is not in the document]
Comment (kr7ysztof) [reply src/main/scala/com/ing/wbaa/rokku/sts/api/xml/TokenXML.scala]: changed

File: src/main/scala/com/ing/wbaa/rokku/sts/data/STSUserInfo.scala [hunk 4]
```diff
@@ -4,8 +4,16 @@ import com.ing.wbaa.rokku.sts.data.aws.{ AwsAccessKey, AwsSecretKey }
...
 case class STSUserInfo(
     userName: UserName,
     userGroup: Set[UserGroup],
     awsAccessKey: AwsAccessKey,
-    awsSecretKey: AwsSecretKey)
+    awsSecretKey: AwsSecretKey,
+    userRole: Option[UserAssumeRole])
+
+sealed trait TokenStatus
+case class TokenActive() extends TokenStatus
```
[the remark this answers is not in the document]
Comment (kr7ysztof) [reply src/main/scala/com/ing/wbaa/rokku/sts/data/STSUserInfo.scala]: changed

**Pull request merged**